server/src/operators/group_operator.rs

```diff
@@ -36,7 +36,11 @@ pub async fn get_group_from_tracking_id_query(
         .filter(chunk_group_columns::tracking_id.eq(tracking_id))
         .first::<ChunkGroup>(&mut conn)
         .await
-        .map_err(|_err| ServiceError::BadRequest("Group not found".to_string()))?;
+        .map_err(|err| {
+            log::error!("Error getting group from tracking_id {:}", err);
+
+            ServiceError::NotFound
```

Comment:
Can you modify the service error so that it returns a message with the 404 so it's more descriptive?

Reply:
Done